inc/c_util/constbuffer.h

```diff
@@ -32,6 +32,25 @@ typedef struct CONSTBUFFER_TAG
 
 typedef void(*CONSTBUFFER_CUSTOM_FREE_FUNC)(void* context);
 
+/*what function should CONSTBUFFER_HANDLE_to_buffer use to allocate the returned serialized form. NULL means malloc from gballoc_hl_malloc_redirect.h of this lib.*/
+typedef void*(*CONSTBUFFER_to_buffer_alloc)(size_t size, void* context);
```

Comment:
Is there a specific use case we need this now? Up to now everything goes through the redirected malloc/free except in very very specific cases like ownership transfer and some THANDLE/COM_WRAPPER cases which are mainly for testing.

If we have a clear use case upstream sure. If we don't and this won't get used then consider not having it.

Reply:
First: specs says you can pass "NULL" and then it uses malloc. That doesn't seem like a huge hinderance to me.

Second: We do not have *at the moment* a use case. However, here's a simple use case: assume you're a receiver of CONSTBUFFER_HANDLEs that need to be serialized in some byte [] array. The size of the byte[] should grow with the CONSTBUFFERs that are received. 

It could be done like 
1. get the serialized size. realloc. serialize in the new grown space. Has a drawback of calling 2 APIs (more mocks!) and sort of does the arithmetic of verification twice.

or (what I had in mind):

2. pass in a custom allocator that does the growth 

err...too much explanation to write here, I've added a sample and will ping you should you want to revisit this topic.

Reply:
Yeah, I get there would be cases of custom allocator that can be used.

The hindrance of passing a NULL is really nothing, the developer can write a NULL when calling this without their hand falling off :-).

What I was challenging is that "if we don't need it now it sounds that maybe we should not add it now".

But it sounds like maybe you are thinking of using this in the near future in the BSDL land, so I'm fine with leaving it in. And probably if not used in a while it will suffer of the "2 are loading, 2 are unloading" syndrome and we just remove it :-).